graylog2-server/src/main/java/org/graylog/scheduler/JobExecutionEngine.java

```diff
@@ -181,4 +196,10 @@ private void executeJob(JobTriggerDto trigger, JobDefinitionDto jobDefinition, J
             }
         }
     }
+
+    private void setupMetrics() {
```

Comment:
If we inline this method and create the metric fields inside the constructor, we could get rid of the `metricRegistry` field. Calling methods inside a constructor can also be tricky because the object might not be completely initialized. If the `setupMetrics()` call would be moved above the `this.metricRegistry = metricRegistry`, the method call would fail.

As an alternative you could also pass in the `metricRegistry` into the `setupMetrics()` method like you did in `JobWorkerPool#registerMetrics`. I usually make these kind of methods that are used in constructors `static` so I cannot use `this.` by accident and thus avoid using a potentially uninitialized field.